server/Makefile

```diff
@@ -100,12 +106,36 @@ ifeq ($(HOME),/)
 	export XDG_CACHE_HOME = /tmp/go-cache/
 endif
 
+OUTPUT_CONFIG ?= config/config.json
+FIPS_IMAGE ?= cgr.dev/mattermost.com/go-msft-fips:1.24-dev
+
 # Go Flags
 GOFLAGS ?= $(GOFLAGS:)
 # We need to export GOBIN to allow it to be set
 # for processes spawned from the Makefile
-export GOBIN ?= $(PWD)/bin
-GO ?= go
+
+ifeq ($(FIPS_ENABLED),true)
+	export GOBIN ?= /root/go/bin
+	GO ?= docker run --rm -v $(PWD)/../../:/build -v $(PWD)/.go:/root/go -v ~/.cache:/.cache -w /build/mattermost/server -e GOFLAGS -e OUTPUT_CONFIG $(FIPS_IMAGE)
```

Comment:
It would be interesting to think about always using Docker to build at some point...